### PR TITLE
created Base Event Processor and Event Registry

### PR DESCRIPTION
**Pull request by yoursweet667**

### CODE REVIEW

File: UNO/src/main/java/ru/yoursweet667/uno/service/event/EventServiceImpl.java [hunk 1]
```diff
@@ -2,27 +2,28 @@
...
-        return null;
+    public void createEvent(String gameId, Event event) {
+        eventHandlerRegistry.getValidator(event).validate(event, gameService.getGame(gameId));
+        eventHandlerRegistry.getProcessor(event).process(event, gameService.getGame(gameId));
     }
 
     @Override
-    public Events getEvents(String gameId, String playerId, Integer fromEventId) {
+    public List<Event> getEvents(String gameId, String playerId, Integer fromEventId) {
+        List<Event> events = gameService.getGame(gameId).getEvents();
```
Comment (phreakiboi) [inline UNO/src/main/java/ru/yoursweet667/uno/service/event/EventServiceImpl.java:25]: NullPointerException if the game is not found by provided gameId

File: UNO/src/main/java/ru/yoursweet667/uno/service/event/processor/BaseEventProcessor.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import ru.yoursweet667.uno.service.model.Game;
+import ru.yoursweet667.uno.service.model.event.Event;
+
+abstract class BaseEventProcessor<T extends Event> implements EventProcessor<T> {
+
+
+    @Override
+    public void process(T event, Game game) {
+        doProcess(event, game);
```
Comment (phreakiboi) [inline UNO/src/main/java/ru/yoursweet667/uno/service/event/processor/BaseEventProcessor.java:11]: this class does nothing

**Review by phreakiboi — CHANGES_REQUESTED** [review]

File: UNO/src/main/java/ru/yoursweet667/uno/service/event/EventServiceImpl.java (hunk 1 above)
Comment (yoursweet667) [reply UNO/src/main/java/ru/yoursweet667/uno/service/event/EventServiceImpl.java:25]: done

File: UNO/src/main/java/ru/yoursweet667/uno/service/event/processor/BaseEventProcessor.java (hunk 2 above)
Comment (yoursweet667) [reply UNO/src/main/java/ru/yoursweet667/uno/service/event/processor/BaseEventProcessor.java:10]: FIxed

**Review by phreakiboi — APPROVED** [review]

**Pull request merged**